test/suites/drift/suite_test.go

```diff
@@ -390,6 +397,12 @@ var _ = Describe("Drift", func() {
 		pod = env.EventuallyExpectHealthyPodCount(selector, numPods)[0]
 		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("ImageId", HaveValue(Equal(amdAMI))))
 
+		By("validating the deprecated status condition has been removed")
+		Eventually(func(g Gomega) {
+			g.Expect(nodeClass.StatusConditions().Clear(v1.ConditionTypeAMIsDeprecated)).To(Succeed())
```

Comment:
Why are we clearing the status condition here?

Reply:
Since the `amiSelectorTerms` are updated, to include a non deprecated AMI, which will update the nodeclass `status.ami` to the non deprecated AMI - causing a drift, which should remove the condition previously added from the deprecated AMI?